packages/database/src/modelClasses/AncestorDescendantRelation.js

```diff
@@ -3,25 +3,59 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { reduceToDictionary } from '@tupaia/utils';
+import { reduceToDictionary, reduceToArrayDictionary } from '@tupaia/utils';
 import { DatabaseModel } from '../DatabaseModel';
 import { DatabaseType } from '../DatabaseType';
 import { TYPES } from '../types';
 
-class AncestorDescendantRelationType extends DatabaseType {
+export class AncestorDescendantRelationType extends DatabaseType {
   static databaseType = TYPES.ANCESTOR_DESCENDANT_RELATION;
+
+  static joins = [
+    {
+      joinWith: TYPES.ENTITY,
+      joinAs: 'descendant',
+      joinCondition: ['descendant_id', 'descendant.id'],
+      fields: { code: 'descendant_code' },
+    },
+    {
+      joinWith: TYPES.ENTITY,
+      joinAs: 'ancestor',
+      joinCondition: ['ancestor_id', 'ancestor.id'],
+      fields: { code: 'ancestor_code' },
+    },
+  ];
 }
 
 export class AncestorDescendantRelationModel extends DatabaseModel {
   get DatabaseTypeClass() {
     return AncestorDescendantRelationType;
   }
 
-  async getChildIdToParentId(hierarchyId) {
-    const relationRecords = await this.find({
+  async getImmediateRelations(hierarchyId) {
+    return this.find({
       entity_hierarchy_id: hierarchyId,
       generational_distance: 1,
     });
+  }
+
+  async getChildIdToParentId(hierarchyId) {
```

Comment:
Will probably refactor out usages of this method in `web-config-server` as I believe they always actually want the `code` instead of `id` anyway...

Reply:
Might hold off on doing this refactor for now as it slightly changes the code logic and there's a bit of complexity there (crossing between `web-config-server` and `web-frontend packages`)

Reply:
No worries, that's sensible